src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,517 @@
+    
+    /**
+     * The key used in the post data to identify nested form data
+     */
+    const POST_KEY = 'GridFieldNestedForm';
+
+    private static $allowed_actions = [
+        'handleNestedItem'
+    ];
+
+    /**
+     * The default max nesting level. Nesting further than this will throw an exception.
+     *
+     * @var boolean
+     */
+    private static $default_max_nesting_level = 10;
+    
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var bool
+     */
+    private $expandNested = false;
+
+    /**
+     * @var bool
+     */
+    private $forceCloseNested = false;
+
+    /**
+     * @var GridField
+     */
+    private $gridField = null;
+
+    /**
+     * @var string
+     */
+    private $relationName = 'Children';
+
+    /**
+     * @var bool
+     */
+    private $inlineEditable = false;
+
+    /**
+     * @var callable|string
+     */
+    private $canExpandCheck = null;
+
+    /**
+     * @var int
+     */
+    private $maxNestingLevel = null;
+    
+    public function __construct($name = 'NestedForm')
+    {
+        $this->name = $name;
+    }
+    
+    /**
+     * Get the grid field that this component is attached to
+     */
+    public function getGridField(): GridField
+    {
+        return $this->gridField;
+    }
+
+    /**
+     * Get the relation name to use for the nested grid fields
+     */
+    public function getRelationName(): string
+    {
+        return $this->relationName;
+    }
+    
+    /**
+     * Set the relation name to use for the nested grid fields
+     */
+    public function setRelationName(string $relationName)
+    {
+        $this->relationName = $relationName;
+        return $this;
+    }
+
+    /**
+     * Get whether the nested grid fields should be inline editable
+     */
+    public function getInlineEditable(): bool
+    {
+        return $this->inlineEditable;
+    }
+
+    /**
+     * Set whether the nested grid fields should be inline editable
+     */
+    public function setInlineEditable(bool $editable)
+    {
+        $this->inlineEditable = $editable;
+        return $this;
+    }
+    
+    /**
+     * Set whether the nested grid fields should be expanded by default
+     */
+    public function setExpandNested(bool $expandNested)
+    {
+        $this->expandNested = $expandNested;
+        return $this;
+    }
+
+    /**
+     * Set whether the nested grid fields should be forced closed on load
+     */
+    public function setForceClosedNested(bool $forceClosed)
+    {
+        $this->forceCloseNested = $forceClosed;
+        return $this;
+    }
+
+    /**
+     * Set a callback function to check which items in this grid that should show the expand link
+     * for nested gridfields. The callback should return a boolean value.
+     * You can either pass a callable or a method name as a string.
+     */
+    public function setCanExpandCheck(callable|string $callback)
+    {
+        $this->canExpandCheck = $callback;
```

Comment:
```suggestion
        $this->canExpandCallback = $callback;
```
As per https://github.com/symbiote/silverstripe-gridfieldextensions/pull/384#discussion_r1573966303